lib/hammer_cli/output/definition.rb

```diff
@@ -8,14 +8,72 @@ def initialize
       @fields = []
     end
 
+    def build(dsl = Dsl.new, &block)
+      dsl.build(&block) if block_given?
+      append(dsl.fields)
+    end
+
     def append(fields)
+      return @fields if fields.nil?
+      fields = [fields] unless fields.is_a? Array
       @fields += fields
     end
 
+    def find_field(field_id)
+      @fields[field_index(field_id)]
+    end
+
+    def insert(mode = :after, field_id = nil, fields = nil, &block)
```

Comment:
How about changing the signature to `def insert(mode, field_id, fields = nil, &block)` ?
Without `field_id` the method will fail anyway so both parameters are required.